scaleway/resource_iam_policy.go

```diff
@@ -0,0 +1,234 @@
+package scaleway
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	iam "github.com/scaleway/scaleway-sdk-go/api/iam/v1alpha1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+func resourceScalewayIamPolicy() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceScalewayIamPolicyCreate,
+		ReadContext:   resourceScalewayIamPolicyRead,
+		UpdateContext: resourceScalewayIamPolicyUpdate,
+		DeleteContext: resourceScalewayIamPolicyDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		SchemaVersion: 0,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Optional:    true,
+				Description: "The name of the iam application",
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The description of the iam application",
+			},
+			"created_at": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The date and time of the creation of the application",
+			},
+			"updated_at": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The date and time of the last update of the application",
+			},
+			"editable": {
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Description: "Whether or not the application is editable.",
+			},
+			"organization_id": organizationIDSchema(),
+			"user_id": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Description:  "User id",
+				ExactlyOneOf: []string{"group_id", "application_id", "no_principal"},
+			},
+			"group_id": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Description:  "Group id",
+				ExactlyOneOf: []string{"user_id", "application_id", "no_principal"},
+			},
+			"application_id": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Description:  "Application id",
+				ExactlyOneOf: []string{"user_id", "group_id", "no_principal"},
+			},
+			"no_principal": {
+				Type:         schema.TypeBool,
+				Optional:     true,
+				Description:  "Application id",
+				ExactlyOneOf: []string{"user_id", "group_id", "application_id"},
+			},
+			"rule": {
```

Comment:
Should it be rules ?

Reply:
it if `rule` so it can be added as singular
```hcl
resource ... {
  rule {
    ...
  }
  rule {
    ...
  }
}
```